### PR TITLE
Create group ArgoCD projects and apps

### PR DESCRIPTION
**Pull request by garryod**
Adds a helm chart which:
- Creates a namespace for each group (with the name `<NAME>-group`)
- Creates an ArgoCD [AppProject](https://argo-cd.readthedocs.io/en/stable/user-guide/projects/) for each group, which can only deploy (Cluster)WorkflowTemplates
- Creates a set of Apps for each group

### CODE REVIEW

File: charts/groups/values.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  server: ""
+
+groups:
+  - name: imaging
+    enabled: true
+    permittedRepos:
+      - https://github.com/DiamondLightSource/imaging-workflows
+    apps:
+      - name: imaging
+        repoURL: https://github.com/DiamondLightSource/imaging-workflows
```
Comment (daurer) [inline charts/groups/values.yaml:12]: can we somehow avoid listing all the repos twice?

File: charts/argocd-apps/values.yaml [hunk 2]
```diff
@@ -64,6 +70,6 @@ sealedsecrets:
 
 workflows:
   enabled: true
-  targetRevision: HEAD
+  targetRevision: app-projects
```
Comment (daurer) [inline charts/argocd-apps/values.yaml:73]: don't forget to clean up before final merge

File: charts/groups/templates/namespaces.yaml
Comment (daurer) [inline charts/groups/templates/namespaces.yaml:1]: what do we need this namespace for?

File: charts/groups/values.yaml (hunk 1 above)
Comment (garryod) [reply charts/groups/values.yaml:12]: Yeah, should be an easy-ish update to the chart. Though maybe we don't even want permitted repos and should just allow them to use whatever?

File: charts/groups/templates/namespaces.yaml
Comment (garryod) [reply charts/groups/templates/namespaces.yaml:1]: Don't absolutely need it, but it makes it easier to write kyverno policy which equates to ArgoCD Signals in the `imaging-group` namespace can create workflows in namespaces x, y & z. Or to give folks from the group permission to modify resources in the namespace

**Review by daurer — APPROVED** [review]

**Pull request merged**